substrate/frame/src/lib.rs

```diff
@@ -0,0 +1,343 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! > Made for [![polkadot]](https://polkadot.network)
+//!
+//! [polkadot]: https://img.shields.io/badge/polkadot-E6007A?style=for-the-badge&logo=polkadot&logoColor=white
+//!
+//! # FRAME
+//!
+//! ```co_compile
+//!   ______   ______    ________   ___ __ __   ______
+//!  /_____/\ /_____/\  /_______/\ /__//_//_/\ /_____/\
+//!  \::::_\/_\:::_ \ \ \::: _  \ \\::\| \| \ \\::::_\/_
+//!   \:\/___/\\:(_) ) )_\::(_)  \ \\:.      \ \\:\/___/\
+//!    \:::._\/ \: __ `\ \\:: __  \ \\:.\-/\  \ \\::___\/_
+//!     \:\ \    \ \ `\ \ \\:.\ \  \ \\. \  \  \ \\:\____/\
+//!      \_\/     \_\/ \_\/ \__\/\__\/ \__\/ \__\/ \_____\/
+//! ```
+//!
+//! > **F**ramework for **R**untime **A**ggregation of **M**odularized **E**ntities: Substrate's
+//! > State Transition Function (Runtime) Framework.
+//!
+//! ## Warning: Experimental
+//!
+//! This crate and all of its content is experimental, and should not yet be used in production.
+//!
+//! ## Getting Started
+//!
+//! TODO: link to `developer_hub::polkadot_sdk::frame`.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![cfg(feature = "experimental")]
+
+/// Exports the main pallet macro. This can wrap a `mod pallet` and will transform it into
+/// being a pallet, eg `#[frame::pallet] mod pallet { .. }`.
+///
+/// Note that this is not part of the prelude, in order to make it such that the common way to
+/// define a macro is `#[frame::pallet] mod pallet { .. }`, followed by `#[pallet::foo]`,
+/// `#[pallet::bar]` inside the mod.
+pub use frame_support::pallet;
+
+/// The logging library of the runtime. Can normally be the classic `log` crate.
+pub use log;
+
+/// The main prelude of FRAME.
+///
+/// This prelude should almost always be the first line of code in any pallet or runtime.
+///
+/// ```
+/// use frame::prelude::*;
+///
+/// // rest of your pallet..
+/// mod pallet {}
+/// ```
+pub mod prelude {
+	/// `frame_system`'s parent crate, which is mandatory in all pallets build with this crate.
+	///
+	/// Conveniently, the keyword `frame_system` is in scope as one uses `use
+	/// frame::prelude::*`
+	#[doc(inline)]
+	pub use frame_system;
+
+	/// Pallet prelude of `frame-support`.
+	///
+	/// Note: this needs to revised once `frame-support` evolves.
```

Comment:
Evolves how? It is unclear to me what this comment is trying to convey

Reply:
`frame-support` will be break down https://github.com/paritytech/polkadot-sdk/issues/127 and its reexports will most likely change.
These frame crate's wildcard reexportings can be optimized once `frame-support` has changed.
I am adding some comments about it